content/docs/source-git/how-to-source-git.md

```diff
@@ -1,218 +1,182 @@
-
-```bash
-$ git remote add upstream-stable https://github.com/systemd/systemd-stable.git
-$ git fetch -q upstream-stable
-```
+At the time of writing this guide:
+* We have chrony 4.0 in Fedora Rawhide
+* 4.0 git tag is also the HEAD of upstream git master (hash:
+  d327cfea5a4b5f7385056be8b18f4c5fab01ad13)
 
-### We can start now
+In order for us to continue, we need to be able to either generate an archive
+from the current HEAD, or download one. This is always going to be specific to
+projects or may depend on what you intend to achieve.
 
-When we open the Fedora 31 systemd spec file, we can see that upstream uses
-version `243.7`. There's a corresponding tag `v243.7` in `upstream-stable`.
-We'll start a new git branch in our repo named `2437-sg` (sg as source-git):
+The use case we cover here is to use a source-git repo for maintenance in
+Fedora Rawhide. This implies we want to download the 4.0 archive from upstream.
+Let's do it!
 
-```bash
-$ git switch -C 2437-sg v243.7
-Switched to a new branch '2437-sg'
+We are going to create a new branch, fedora-rawhide, to be the source-git for
+rawhide.
 ```
-
-Right now we have the upstream history which lead to the 243.7 release.
-Alternatively we could just unpack the upstream tarball and have the history in
-a single commit.
-
-### Layering downstream content on top of upstream
-
-Let's get files from the `fedora/f31` branch.
-
-```bash
-$ git checkout fedora/f31 -- .
-$ git status
-On branch 2437-sg
-Changes to be committed:
-  (use "git restore --staged <file>..." to unstage)
-	modified:   .gitignore
-	new file:   0002-Revert-units-set-NoNewPrivileges-for-all-long-runnin.patch
-	new file:   0998-resolved-create-etc-resolv.conf-symlink-at-runtime.patch
-	new file:   20-grubby.install
-	new file:   20-yama-ptrace.conf
-	new file:   464a73411c13596a130a7a8f0ac00ca728e5f69e.patch
-	new file:   inittab
-	new file:   purge-nobody-user
-	new file:   sources
-	new file:   split-files.py
-	new file:   sysctl.conf.README
-	new file:   systemd-journal-gatewayd.xml
-	new file:   systemd-journal-remote.xml
-	new file:   systemd-udev-trigger-no-reload.conf
-	new file:   systemd-user
-	new file:   systemd.spec
-	new file:   triggers.systemd
-	new file:   yum-protect-systemd.conf
+$ git checkout -B fedora-rawhide 4.0
 ```
 
-We have a bunch of new files from f31 branch in the root.
-We'll move them now to the `fedora/` directory:
 
-```bash
-$ mkdir fedora
-$ mv $(git diff --name-only --cached) fedora/
-```
+## RPM spec file
 
-and we should also clean our working tree:
+It's most convenient to have the spec file stored in the source-git repo since
+you are in full control of everything.
 
-```bash
-$ git reset HEAD .
-Unstaged changes after reset:
-M       .gitignore
+Here lies another decision for you: to pick the source of your spec file.
 
-$ git restore .gitignore
+Since our use case is clear, Fedora Rawhide package maintenance, we'll gonna
+pick our spec file from rawhide:
```

Comment:
```suggestion
pick our spec file from [rawhide](https://src.fedoraproject.org/rpms/chrony/tree/master):
```

Reply:
well, the link is available below though this is explicit